pom.xml

```diff
@@ -30,7 +30,27 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
-        <embedded-postgres-binaries.version>14.10.1</embedded-postgres-binaries.version>
+        <embedded-postgres-binaries.version>14.11.0</embedded-postgres-binaries.version>
+        <commons-codec.version>1.17.0</commons-codec.version>
+        <commons-compress.version>1.26.2</commons-compress.version>
+        <commons-io.version>2.16.1</commons-io.version>
+        <commons-lang3.version>3.14.0</commons-lang3.version>
+        <flyway.version>7.15.0</flyway.version>
+        <junit5.version>5.10.3</junit5.version>
+        <junit4.version>4.13.2</junit4.version>
+        <liquibase.version>4.28.0</liquibase.version>
+        <mockito.version>4.11.0</mockito.version>
+        <pmd.version>7.3.0</pmd.version>
+        <postgresql.version>42.7.3</postgresql.version>
+        <slf4j.version>2.0.13</slf4j.version>
```

Comment:
I'm afraid this change is not entirely backwards compatible. The versions of slf4j-api and slf4j-simple must match and this cannot be enforced within this library. So we have to stay with version 1.7.36.

https://www.slf4j.org/faq.html#changesInVersion200

Reply:
I'm not sure I follow, you can see that I changed your dependency management to import the [slf4j-bom](https://mvnrepository.com/artifact/org.slf4j/slf4j-bom), so all slf4j dependencies are using the same version automatically... the power of [BOM](https://www.baeldung.com/spring-maven-bom), I like them 😅

Reply:
The thing is that slf4j-bom ensures consistent versions only within this project (during library build, running its tests, etc.). But when I add this library as a Maven or Gradle dependency to another project, this BOM will no longer have an effect on that project.

`slf4j-bom` - has no effect on the target project
`slf4j-api` - compile scoped - can affect the version in the target project as a transitive dependency
`slf4j-simple` - test scoped - cannot affect the target project

So if someone doesn't have versions fixed through dependency management (yes, it's unwise, but it can happen), this change is potentially backwards incompatible.

Reply:
Still not following, a BOM is not introducing a problem for me 😅
I just tested it in a lib and an app project.
```xml
<!-- lib -->
<properties>
  <slf4j.version>2.0.13</slf4j.version>
</properties>
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-bom</artifactId>
      <version>${slf4j.version}</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
<dependencies>
  <dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-api</artifactId>
  </dependency>
  <dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-simple</artifactId>
    <scope>test</scope>
  </dependency>
</dependencies>
```
```
--- dependency:3.7.0:tree (default-cli) @ lib-with-slf4j-bom ---
io.example:lib-with-slf4j-bom:jar:1.0-SNAPSHOT
+- org.slf4j:slf4j-api:jar:2.0.13:compile
\- org.slf4j:slf4j-simple:jar:2.0.13:test
```
```xml
<!-- app -->
<dependencies>
  <dependency>
    <groupId>io.example</groupId>
    <artifactId>lib-with-slf4j-bom</artifactId>
    <version>1.0-SNAPSHOT</version>
  </dependency>
  <dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-api</artifactId>
  </dependency>
</dependencies>
```

```xml
<!-- app -->
<dependencies>
  <dependency>
    <groupId>io.example</groupId>
    <artifactId>lib-with-slf4j-bom</artifactId>
    <version>1.0-SNAPSHOT</version>
  </dependency>
  <!-- note: no explicit slf4j-api dependency -->
</dependencies>
```
```
--- dependency:3.7.0:tree (default-cli) @ app-depending-on-slf4j-bom-lib ---
io.example:app-depending-on-slf4j-bom-lib:jar:1.0-SNAPSHOT
\- io.example:lib-with-slf4j-bom:jar:1.0-SNAPSHOT:compile
   \- org.slf4j:slf4j-api:jar:2.0.13:compile
```
So the app could import slf4j 2.0.13 now in Java classes.
If the app would now add slf4j-api explicitly (which should be best practice if you import external classes), a specific version would be required (maybe through a BOM import). Please note that the slf4j.version property was not transitive, so it couldn't be used here.
slf4j-simple would need a version too (not transitive).

So to me the app always wins, as you can't rely on all dependencies managing the exact same transitive versions you need in your project... so I always manage any version I need myself... so my code can compile? 😅

I could make it `<optional>true</optional>` (not transitive), it's compiled anyway.
Or revert it to the old version, your call 😄

Reply:
I'm concerned about the following scenario:

```
<!-- app -->
<dependencies>
  <dependency>
    <groupId>io.example</groupId>
    <artifactId>lib-with-slf4j-bom</artifactId>
    <version>1.0-SNAPSHOT</version>
  </dependency>
  <dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-simple</artifactId>
    <version>1.7.36</version>
  </dependency>
</dependencies>
```

In my opinion, in such a case, it could happen that the version of slf4j-api is 2.0.13, while slf4j-simple is 1.7.36. If so, this is potentially a backward incompatible change. And according to semantic versioning, such a change should be released with the next major version.

Can you try it?

Reply:
Yes, the versions would diverge if the user would not manage those dependencies himself.
```
--- dependency:3.7.0:tree (default-cli) @ app-depending-on-slf4j-bom-lib ---
io.example:app-depending-on-slf4j-bom-lib:jar:1.0-SNAPSHOT
+- io.example:lib-with-slf4j-bom:jar:1.0-SNAPSHOT:compile
|  \- org.slf4j:slf4j-api:jar:2.0.13:compile
\- org.slf4j:slf4j-simple:jar:1.7.36:compile
```
There is no BOM for slf4j < 2.0.8, so I will revert that too.